lib/bowling.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require_relative './frame'
+
+class Bowling
+  attr_reader :total_score
+
+  def initialize(input: $stdin, output: $stdout)
+    @frames = []
+    @total_score = 0
+    @input = input
+    @output = output
+  end
+
+  def play
+    (1..10).each do |i|
+      @current_frame = Frame.new
+      2.times { game_roll }
+      process_frame(i)
+      p @current_frame.result unless @current_frame.result.empty?
+    end
+    p "total score: #{@total_score}"
+  end
+
+  private
```

Comment:
Nice use of using private to separate bonus rolls and previous frame scores. I hadn't thought of this! I assume this is so the user can't access these to alter to the scores and affect the final scorecard?